### DISCUSSION TITLE
Compression artifacts processed as full frames of motion?

### DISCUSSION THREAD

**Discussion by mac587** [blakeblackshear/frigate#8138] [Ask A Question]
On a calm day, I'm seeing a loop where lots of small motion boxes eventually extend out to entire frames, when this happens there seems to be a small delay before lots of small motion boxes appear everywhere, again becoming larger to full frames. This repeats constantly.

Detect CPU% for the cams in question is also very high compared to others (10-20% of i5-6500)

I understand  `max_area` setting has no effect on motion (unlike motioneye which I used previously)

Frigate is seemingly constantly processing many large frames for no reason other than very lossy compression artifacts/noise, I think possibly made worse by large area of gravel.

Cameras are tapo c310v2, using main 2k stream for detect as sub stream quality is too low for far away objects in this scenario

Any ideas if this expected behavior or I missing something?

Perhaps related to `improve_contrast: True` ?

````
mqtt:
  host: 192.168.1.19
  port: 1883
  user: redacted
  password: redacted

ffmpeg:
  hwaccel_args: preset-vaapi
 
birdseye:
  enabled: True
  mode: continuous

detectors:

  coral_pci:
    type: edgetpu
    device: pci

record:
  enabled: True
  retain:
    days: 14
    mode: motion
  events:
    retain:
      default: 30
      mode: active_objects

snapshots:
  enabled: True

#Global Object Settings

objects:
  track:
    - person
    - dog
    - cat
    - bird
    - horse
    - car
  filters:
    person:
      threshold: 0.74
    dog:
      threshold: 0.65
    cat:
      threshold: 0.8
    bird:
      threshold: 0.8
    horse:
      threshold: 0.8
    car:
      threshold: 0.77


motion:

  improve_contrast: True
 
go2rtc:

  streams:

     parking:
      - rtsp://redacted:redacted@192.168.1.2:554/stream1
      - "ffmpeg:parking#audio=aac" # <- copy of the stream which transcodes audio to opus
     parking_sub:
      - rtsp://redacted:redacted@192.168.1.2:554/stream2

cameras:

  parking:
    ffmpeg:
      output_args:
        record: preset-record-generic-audio-copy
      inputs:
        - path: rtsp://127.0.0.1:8554/parking
          input_args: preset-rtsp-restream
          roles:   
            - detect
            - record
    detect:
      width: 1366
      height: 768
      fps: 5

**Comment by NickM-27** [comment]
I think an example of the debug view with motion boxes would be helpful to see. Motion is pixels changing, so if there is a high level of compression artifacts that could definitely be detected as motion if it is too drastic. Frigate of course has configurable motion detection so it could be made less sensitive.

**Reply by mac587** [reply]
Ah I see, 
`motion:
 threshold: ` must be what I'm looking for
I'm having trouble getting screenshots of what's happening, but starts out as many small boxes all over the image, which seem to get larger and start overlapping, to one big full frame box, to nothing, rinse and repeat

**Reply by kirsch33** [reply]
https://github.com/blakeblackshear/frigate/discussions/7850

incase you didnt know it existed

**Reply by hawkeye217** [reply]
When Frigate detects motion in a large percentage of the frame, the motion detector can trip into a "calibrating" mode where Frigate is trying to establish a baseline or reference state for motion detection. This mode helps differentiate between normal motion (like a person moving in the frame) and changes in the scene that are not relevant for motion detection (like changes in lighting conditions).

As @NickM-27 said, compression artifacts are likely creating this excess motion. You could increase the bitrate of your camera stream (which would likely produce less compression artifacts) and/or tweak the motion parameters.

**Reply by NickM-27** [reply]
@hawkeye217 youre assuming OP is running the beta, I'm not sure that's the case

**Reply by hawkeye217** [reply]
You speak truth, @NickM-27. Though I don't recall ever seeing the behavior OP describes with the motion detector in 0.12.

**Reply by NickM-27** [reply]
yeah, I think it is difficult to know what is going on without an example, compression / other artifacting can take many different forms

**Reply by mac587** [reply]
Thanks guys, yes this is on 0.12. 
 Changing motion threshold from the default (25) to 35 tamed things down a lot, yet was still seeing many smaller boxes on the gravel. 
40 seems to have solved it completely while still getting reliable object detection.

 Unfortunately these cams ship with locked ONVIF settings of "quality: 3" "bitrate :1536" on the main 2k stream, but they're 1/3rd the price of cams on the recommended list so can't complain really

Thanks again
